public/pages/workflow_detail/agentic_search/configure_flow/configure_flow.tsx

```diff
@@ -0,0 +1,276 @@
+  const selectedAgentId = getIn(values, AGENT_ID_PATH, '') as string;
+  // TODO: long-term, wrap all agent fields in a nested formik form for consistent validation/touch/submit actions
+  const [agentForm, setAgentForm] = useState<Partial<Agent>>(EMPTY_AGENT);
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+
+  // persist unsaved states when users are creating new / updating existing agents.
+  // keep these states as 'true' if errors when trying to execute the respective
+  // create/update agent API calls.
+  const [newAndUnsaved, setNewAndUnsaved] = useState<boolean>(false);
+  const [existingAndUnsaved, setExistingAndUnsaved] = useState<boolean>(false);
+  useEffect(() => {
+    setExistingAndUnsaved(
+      !isEmpty(selectedAgentId) &&
+        selectedAgentId !== NEW_AGENT_PLACEHOLDER &&
+        !isEqual(getIn(agents, selectedAgentId, {}), agentForm)
+    );
+  }, [selectedAgentId, agents, agentForm]);
+  const unsaved = newAndUnsaved || existingAndUnsaved;
+  const createOrUpdateDisabled =
+    isEqual(getIn(agents, selectedAgentId, {}), agentForm) ||
+    isEmpty(selectedAgentId) ||
+    isEmpty(agentForm?.name) ||
+    (agentForm.type !== AGENT_TYPE.FLOW && isEmpty(agentForm.llm?.model_id)) ||
+    (agentForm.type !== AGENT_TYPE.FLOW &&
+      isEmpty(agentForm.parameters?._llm_interface)) ||
+    (!isEmpty(
+      agentForm?.tools?.find((tool) => tool.type === TOOL_TYPE.QUERY_PLANNING)
+    ) &&
+      isEmpty(
+        agentForm?.tools?.find((tool) => tool.type === TOOL_TYPE.QUERY_PLANNING)
+          ?.parameters?.model_id
+      ));
+
+  // fine-grained error handling states
+  const [errorCreatingAgent, setErrorCreatingAgent] = useState<string>('');
+  const [errorUpdatingAgent, setErrorUpdatingAgent] = useState<string>('');
+  function clearAgentErrors() {
+    setErrorCreatingAgent('');
+    setErrorUpdatingAgent('');
+  }
+
+  function onCreateNew() {
+    setNewAndUnsaved(true);
+    clearAgentErrors();
+  }
+  function onDiscardDraft() {
+    clearAgentErrors();
+    setNewAndUnsaved(false);
+    setFieldValue(
+      AGENT_ID_PATH,
+      props.uiConfig?.search?.requestAgentId?.value || undefined
+    );
+  }
+  function onRevertChanges() {
+    clearAgentErrors();
+    const agent = getIn(agents, selectedAgentId, {});
+    if (!isEmpty(agent)) {
+      setAgentForm(agent);
+    } else {
+      setAgentForm(EMPTY_AGENT);
+    }
+  }
+  async function onCreateAgent() {
+    setIsSaving(true);
+    let errorOnCreate = false;
+    try {
+      const response = await dispatch(
+        registerAgent({
+          apiBody: agentForm,
+          dataSourceId,
+        })
+      ).unwrap();
+      if (response && response.agent && response.agent.id) {
+        setFieldValue(AGENT_ID_PATH, response.agent.id);
+      }
+      clearAgentErrors();
+    } catch (error) {
+      errorOnCreate = true;
+      setErrorCreatingAgent(error);
```

Comment:
Do we need to type cast the error?
```suggestion
      setErrorCreatingAgent(String(error));
```

Reply:
sure will add.